pkg/controllers/federate/util.go

```diff
@@ -71,12 +73,17 @@ func newFederatedObjectForSourceObject(
 		[]metav1.OwnerReference{*metav1.NewControllerRef(sourceObj, sourceObj.GroupVersionKind())},
 	)
 
-	federatedAnnotations, templateAnnotations := classifyAnnotations(sourceObj.GetAnnotations())
-	fedObj.SetAnnotations(federatedAnnotations)
-
 	federatedLabels, templateLabels := classifyLabels(sourceObj.GetLabels())
 	fedObj.SetLabels(federatedLabels)
 
+	observedLabels := generateObservedLabels(sourceObj.GetLabels())
```

Comment:
Why don't we pass `federatedLabels` into `generateObservedLabels`? In the future the implementation of `classifyLabels` might change, and `generateObservedLabels` should not rely on its current implementation.

Reply:
With this change, `generateObservedAnnotations` and `generateObservedLabels` will be identical. We can keep only one of them.